### ISSUE TITLE
Split `IBitcoindRpc` into separate server-side and client-side traits

### ISSUE THREAD

**Issue by justinmoon** [fedimint/fedimint#3331]
`IBitcoindRpc` contains methods that aren't supposed to be available on the fedimint servers such as `get_txout_proof`. Consider splitting this into separate traits for client and server.

**Comment by elsirion** [comment]
If someone takes this on and has trouble understanding the `Dyn*` pattern ping me or @dpc.

**Comment by dpc** [comment]
For the RBF UTXO management we might need `get_txout_proof` for last few blocks (pruned node) for server side. @bradleystachurski , not sure if there's an issue for that already.

**Comment by bradleystachurski** [comment]
@dpc Nope, no issue yet. I'm planning on creating issues for the RBF work once everyone has a chance to review the proposal and agrees it's a sane approach.

If nobody else grabs this, I'll pick it up. For visibility, I won't have the bandwidth until early November but if nobody else has started working on this I'll self assign.

**Comment by 15IITian** [comment]
Hey @bradleystachurski , are you still working on it?

**Comment by bradleystachurski** [comment]
Hey @15IITian, thanks for checking! I'm not currently working on this so feel free to grab it.

**Comment by 15IITian** [comment]
> @15IITian The initial issue description is clear. If that doesn't make sense, this probably isn't a good issue for you to work on. I'm removing the "good first issue" label because this requires some context about Fedimint architecture.

OK

**Comment by elsirion** [comment]
Closed by a bunch of recent @joschisan PRs
